### PR TITLE
Search filters

### PR DESCRIPTION
**Pull request by tigermli**
- Search Queries now separated by spaces and searched individually. Final result is intersection of all.
- Parent model removed from Search. Admin added to serializer and view.
- Filter names changed:
Account Search Filters: profile, grade, sort, page
Course Search Filters: course, availability ,sort, page

### CODE REVIEW

File: search/views.py [hunk 1]
```diff
@@ -22,47 +26,84 @@ def get_queryset(self):
...
-                            searchResults = searchResults.filter(grade=gradeFilter)
-                    except ValueError:
-                        pass
-            elif profileFilter == "Instructor":
-                searchResults = Instructor.objects.search(query)
-            elif profileFilter == "Admin":
-                searchResults = Admin.objects.search(query)
+                try:
+                    gradeFilter = int(self.request.query_params.get('grade', None))
+                    if 1 <= gradeFilter and gradeFilter <= 12:
```
Comment (jerrylinew) [inline search/views.py]: Grade can be 13 (graduated)

**Review by jerrylinew — APPROVED** [review]
LGTM. Have you tested it out with all of summit's data loaded? How long does search take?

**Comment by tigermli** [comment]
> LGTM. Have you tested it out with all of summit's data loaded? How long does search take?

With the page number filter the search is pretty fast on Summit data since only 8 results are returned at a time!

File: search/views.py (hunk 1 above)
Comment (tigermli) [reply search/views.py]: Will fix that

**Pull request merged**